### PR TITLE
[Scan to update inventory] Part 5 - analytics events

### LINKED ISSUE #10338: [Scan to update inventory] Part 5 - Analytics events

### PR DESCRIPTION
**Pull request by rossanafmenezes**
Closes: #10338 

### Description

This PR adds the tracking events for the Scan To Update Inventory Project.

### Testing instructions

- Go to Product list screen, and tap the “Scan barcode” icon —  make sure `product_list_product_barcode_scanning_tapped ` is tracked
- Scan a barcode that is not registered on your store  — make sure `product_search_via_sku_failure` with source property `scan_to_update_inventory` is tracked. 
 - Scan an existing product - make sure `product_search_via_sku_successwith` the source property `scan_to_update_inventory` is tracked
 - Scan an existing product, then click on the button to increment the quantity - make sure  `product_quick_inventory_update_increment_quantity_tapped` is tracked
 - scan an existing product, then increase the quantity manually and tap on the update quantity button. Make sure `product_quick_inventory_update_manual_quantity_update_tapped` is tracked
 - Scan an existing product. Dismiss the bottom sheet. Make sure `product_quick_inventory_update_dismissed` is tracked. 
 - Scan existing product. Update quantity and trigger failure. Make sure `product_quick_inventory_quantity_update_failure` is tracked
 - Scan existing product. Tap on `view product details`  button. Make sure `product_quick_inventory_view_product_details_tapped` is tracked. 
 - after the quantity is successfully updated, make sure `product_quick_inventory_quantity_update_success` is tracked
### Images/gif

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt [hunk 1]
```diff
@@ -75,6 +80,13 @@ class ScanToUpdateInventoryViewModel @Inject constructor(
             } else {
                 handleProductNotFound(status.code)
             }
+
+            tracker.track(
+                AnalyticsEvent.PRODUCT_SEARCH_VIA_SKU_SUCCESS,
+                mapOf(
+                    AnalyticsTracker.SCAN_TO_UPDATE_INVENTORY to status.code
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt]:
```suggestion
                    KEY_SCANNING_SOURCE to AnalyticsTracker.SCAN_TO_UPDATE_INVENTORY
```
The prop's key should be equal to "source", and value "scan_to_update_inventory"

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt [hunk 2]
```diff
@@ -95,6 +107,14 @@ class ScanToUpdateInventoryViewModel @Inject constructor(
     }
 
     private suspend fun handleProductNotFound(barcode: String) {
+        tracker.track(
+            AnalyticsEvent.PRODUCT_SEARCH_VIA_SKU_FAILURE,
+            mapOf(
+                AnalyticsTracker.SCAN_TO_UPDATE_INVENTORY to resourceProvider.getString(
+                    R.string.scan_to_update_inventory_unable_to_find_product, barcode
+                )
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt]:
```suggestion
                KEY_SCANNING_SOURCE to AnalyticsTracker.SCAN_TO_UPDATE_INVENTORY
```
The prop's key should be equal to "source", and value "scan_to_update_inventory"

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt [hunk 3]
```diff
@@ -134,11 +156,13 @@ class ScanToUpdateInventoryViewModel @Inject constructor(
                 product.updateProduct(updatedProductInfo)
             }
             if (result.isSuccess) {
+                AnalyticsTracker.track(AnalyticsEvent.PRODUCT_QUICK_INVENTORY_QUANTITY_UPDATE_SUCCESS)
                 handleQuantityUpdateSuccess(
                     product.stockQuantity.toInt().toString(),
                     updatedProductInfo.quantity.toString()
                 )
             } else {
+                AnalyticsTracker.track(AnalyticsEvent.PRODUCT_QUICK_INVENTORY_QUANTITY_UPDATE_FAILURE)
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt]: Let's move it into the `handleQuantityUpdateError()` body – this function is called in two places and we want to track this event each time.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModelTest.kt [hunk 4]
```diff
@@ -359,4 +363,31 @@ class ScanToUpdateInventoryViewModelTest : BaseUnitTest() {
         val expectedVariation = originalVariation.copy(stockQuantity = (999).toDouble())
         verify(variationRepo).updateVariation(expectedVariation)
     }
+
+    @Test
+    fun `when increment quantity button tapped, then proper tracking event is triggered`() = testBlocking {
+
```
Comment (samiuelson) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModelTest.kt]:
```suggestion
```

**Review by samiuelson — COMMENTED** [review]
Looks good overall, @rossanafmenezes!

There are a few small issues related to event props - take a look

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModelTest.kt (hunk 4 above)
Comment (rossanafmenezes) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModelTest.kt]: Fixed here https://github.com/woocommerce/woocommerce-android/pull/10339/commits/5d664568aab9aebc20ae649e4c01a6208de59710

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt (hunk 1 above)
Comment (rossanafmenezes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt]: Sorry about that. Fixed here https://github.com/woocommerce/woocommerce-android/pull/10339/commits/4ebb6f1ad3041450a1bd4394980692d0183d7928

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt (hunk 2 above)
Comment (rossanafmenezes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt]: same as above https://github.com/woocommerce/woocommerce-android/pull/10339/commits/4ebb6f1ad3041450a1bd4394980692d0183d7928

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt (hunk 3 above)
Comment (rossanafmenezes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/inventory/ScanToUpdateInventoryViewModel.kt]: Fixed here https://github.com/woocommerce/woocommerce-android/pull/10339/commits/d8d4d3f54d4d43909efb7c258de6402422e474c2

**Review by samiuelson — APPROVED** [review]
LGTM

**Pull request merged**